apps/cowswap-frontend/src/modules/swap/services/ethFlow/steps/signEthFlowOrderStep.ts

```diff
@@ -58,21 +58,29 @@ export async function signEthFlowOrderStep(
   }
 
   const ethTxOptions = { value: etherValue.quotient.toString() }
-  const estimatedGas = await ethFlowContract.estimateGas
-    .createOrder(ethOrderParams, { value: etherValue.quotient.toString() })
-    .catch((error) => {
-      logTradeFlowError(
-        'ETH FLOW',
-        '[EthFlow::SignEthFlowOrderStep] Error estimating createOrder gas. Using default ' + GAS_LIMIT_DEFAULT,
-        error
-      )
-      return GAS_LIMIT_DEFAULT
-    })
-
-  const txReceipt = await ethFlowContract.createOrder(ethOrderParams, {
+  const estimatedGas = await ethFlowContract.estimateGas.createOrder(ethOrderParams, ethTxOptions).catch((error) => {
+    logTradeFlowError(
+      'ETH FLOW',
+      '[EthFlow::SignEthFlowOrderStep] Error estimating createOrder gas. Using default ' + GAS_LIMIT_DEFAULT,
+      error,
+    )
+    return GAS_LIMIT_DEFAULT
+  })
+
+  // This used to be done with a higher level of abstraction like this:
```

Comment:
The comment is a bit verbose, which I don't mind, but maybe can start giving some context. What do you think about something like:

Ensure theEth flow contract network matches the network where you place the transaction. Ther's multiple wallet implementation, and potential race conditions that can cause the the chain of the wallet to be different, and therefore leaving the chainId implicit might lead the user to place an order in an unwanted chain. This is especially dangerous for Eth Flow orders, because the contract address is different for the distinct networks, and this can lead to lose of funds.

Reply:
https://github.com/cowprotocol/cowswap/pull/5263